src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let currentState = {};
+
+  for (const key in state) {
+    if (state.hasOwnProperty(key)) {
+      currentState[key] = state[key];
+    }
+  }
```

Comment:
Instead of manually copying properties from the `state` object, you can use the spread operator for a more concise and readable approach, like so: `let currentState = { ...state };`.